### PR TITLE
Store volume level on window object

### PR DESCRIPTION
**Pull request by ismailkarsli**
## Description

Currently every video starts muted. With this PR, the volume value is kept in the window and used while playing other videos.

#### componentDidMount is formatted

Husky auto formatting line 131 while commiting.

### CODE REVIEW

**Comment by alectrocute** [comment]
I'd suggest using `window.localStorage.[getItem/setItem]` instead to persist. Let's avoid adding things to the `window` object.

**Comment by ismailkarsli** [comment]
> I'd suggest using `window.localStorage.[getItem/setItem]` instead to persist. Let's avoid adding things to the `window` object.

I specifically did it this way so that the volume value starts muted in each session. However, if you suggest otherwise, I can do it with localStorage (or another global store if available).

**Comment by ismailkarsli** [comment]
Alright I've changed it to work with localStorage.

File: src/shared/components/post/post-listing.tsx [hunk 1]
```diff
@@ -214,7 +214,15 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
     if (url && isVideo(url)) {
       return (
         <div className="embed-responsive ratio ratio-16x9 mt-3">
-          <video muted controls className="embed-responsive-item col-12">
+          <video
+            onLoadStart={this.handleVideoLoadStart}
```
Comment (SleeplessOne1917) [inline src/shared/components/post/post-listing.tsx:218]: Please use `linkEvent` for these as described in [inferno's docs](https://www.infernojs.org/docs/guides/event-handling).

File: src/shared/components/post/post-listing.tsx [hunk 2]
```diff
@@ -766,6 +774,21 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
...
+    if (volume) video.volume = Number(volume);
+    else {
+      video.volume = 0;
+      localStorage.setItem("volume_level", "0");
+    }
+  }
+
+  handleVideoVolumeChange(e) {
+    const video = e.target as HTMLVideoElement;
+    localStorage.setItem("volume_level", String(video.volume));
```
Comment (SleeplessOne1917) [inline src/shared/components/post/post-listing.tsx:789]: Nitpick: can use `toString` inatead.

**Review by SleeplessOne1917 — CHANGES_REQUESTED** [review]
Besides the changes requested from my other comments, please run prettier. It looks like some formatting changed.

**Comment by ismailkarsli** [comment]
> Besides the changes requested from my other comments, please run prettier. It looks like some formatting changed.

I've run prettier (in fact it runs with pre-commit hook anyways) but that part of the code still stays formatted. I guess it was not formatted according to the config before.

image

File: src/shared/components/post/post-listing.tsx [hunk 3]
```diff
@@ -128,10 +128,10 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
 
   componentDidMount(): void {
     if (UserService.Instance.myUserInfo) {
-      const { auto_expand, blur_nsfw } = UserService.Instance.myUserInfo.local_user_view.local_user;
+      const { auto_expand, blur_nsfw } =
```
Comment (dessalines) [inline src/shared/components/post/post-listing.tsx:131]: Hrm, weird how this is getting reformatted. As long as its passing the CI prettier check it should be fine tho.

**Review by dessalines — APPROVED** [review]

**Comment by ismailkarsli** [comment]
I've checked it and event.target is undefined when used with `linkEvent`. Lemmy fix that

File: src/shared/components/post/post-listing.tsx [hunk 4]
```diff
@@ -766,6 +774,21 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
...
+      video.volume = 0;
+      localStorage.setItem("volume_level", "0");
+    }
+  }
+
+  handleVideoVolumeChange(e) {
+    const video = e.target as HTMLVideoElement;
+    localStorage.setItem("volume_level", video.volume.toString());
+  }
+
```
Comment (SleeplessOne1917) [inline src/shared/components/post/post-listing.tsx:791]: These could be defined at the module level instead of as properties of the component.

Comment (ismailkarsli) [reply src/shared/components/post/post-listing.tsx:791]: You mean somewhere like `src/shared/utils/media`?

Comment (SleeplessOne1917) [reply src/shared/components/post/post-listing.tsx:791]: I mean in the same file they're already in, but in the outermost scope.

File: src/shared/components/post/post-listing.tsx [hunk 5]
```diff
@@ -214,7 +214,15 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
     if (url && isVideo(url)) {
       return (
         <div className="embed-responsive ratio ratio-16x9 mt-3">
-          <video muted controls className="embed-responsive-item col-12">
+          <video
+            onLoadStart={linkEvent(this, this.handleVideoLoadStart)}
+            onPlay={linkEvent(this, this.handleVideoLoadStart)}
+            onMouseOver={linkEvent(this, this.handleVideoLoadStart)}
+            onFocus={linkEvent(this, this.handleVideoLoadStart)}
+            onVolumeChange={linkEvent(this, this.handleVideoVolumeChange)}
```
[the remark this answers is not in the document]
Comment (dessalines) [reply src/shared/components/post/post-listing.tsx:222]: Either one works... its not a bad practice to just always use linkEvent, even if you don't need the component.

File: src/shared/components/post/post-listing.tsx (hunk 4 above)
Comment (ismailkarsli) [reply src/shared/components/post/post-listing.tsx]: With the recent changes, I brought back `linkEvent` as @dessalines suggested, should I still move?

**Review by SleeplessOne1917 — APPROVED** [review]

**Pull request merged**